extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -493,6 +494,13 @@ export class SchemaCompareResult {
 						'startTime': Date.now().toString(),
 						'operationId': this.comparisonResult.operationId
 					});
+
+					// disable apply and generate script buttons because the results are no longer valid after applying the changes
+					this.generateScriptButton.enabled = false;
+					this.generateScriptButton.title = reCompareToRefeshMessage;
+					this.applyButton.enabled = false;
+					this.applyButton.title = reCompareToRefeshMessage;
+
 					const service = await SchemaCompareResult.getService('MSSQL');
 					const result = await service.schemaComparePublishChanges(this.comparisonResult.operationId, this.targetEndpointInfo.serverName, this.targetEndpointInfo.databaseName, azdata.TaskExecutionMode.execute);
 					if (!result || !result.success) {
```

Comment:
If apply failed... we should allow second apply or generate script. You can do that by either changing to recompare only after apply result comes back or by changing back if it failed...  whatever looks better on UI

Reply:
hmm I'll add that it the buttons will be enabled again if it wasn't a success, but I think most failures would be returned from the task and not where this code is, so the buttons would still be disabled in that case.